content/js/getting-started.md

```diff
@@ -2,39 +2,92 @@
+a promise object:
 
-- Authenticating users
-- Saving/Querying records to/from the database with various data types
-- Setting access control for different users and roles in accessing
-  database records
-- Managing follower/following/friend relationship between users
-- Sending and receiving instant message via Publish-Subscribe (PubSub)
-- Uploading images and files
+- **Create / Authenticate Users** (see [User & Authentication]())
+```js
+skygear.signupWithUsername('alice', 'P@s$w0Rd'); // create user
+skygear.loginWithUsername('alice', 'P@s$w0Rd'); // login
+```
+
+- **Save Records** (see [Records]())
+```js
+const PersonRecord = skygear.Record.extend('person');
+const bob = new PersonRecord({ name: 'bob' });
+skygear.publicDB.save(bob);
+```
+
+- **Query Records** (see [Queries]())
+```js
+const PersonRecord = skygear.Record.extend('person');
+const findBob = new skygear.Query(PersonRecord).equalTo('name', 'bob');
+skygear.publicDB.query(findBob);
+```
+
+- **Send / Revieve Message via PubSub Channels** (see [PubSub Channels]())
+```js
+skygear.on('message', data => console.log(data)); // subscribe on channel
+skygear.pubsub.publish('message', { hello: 'world' }); // publish to channel
+```
+
+- **Manage Access Control** (see [Access Control]())
+```js
+const bob = new skygear.User({ user_id: '523e7df5-655c-4250-8df0-c38271ea53ba' });
+const MessageRecord = skygear.Record.extend('message');
+const message = new MessageRecord({ something: 'bad about Bob' });
+message.setNoAccessForUser(bob); // to talk behind Bob's back
+skygear.publicDB.save(message);
+```
+
+- **Perform Geolocational Queries** (see [Geolocation]())
+```js
+const BuildingRecord = skygear.Record.extend('building');
+const saxonTower = new BuildingRecord({ location: new skygear.Geolocation(22.3366140, 114.1486090) });
+const leroyPlaza = new BuildingRecord({ location: new skygear.Geolocation(22.3361170, 114.1474730) });
+skygear.publicDB.save([saxonTower, leroyPlaza]);
+// ----------------------------------------------------------------------
+const ourskyOffice = new skygear.Geolocation(22.3362650, 114.1479320);
+const buildingsNearOursky = new skygear.Query(BuildingRecord)
+    .distanceLessThan('location', me, 1000); // all buildings within 1km radius
+skygear.publicDB.query(buildingsNearOursky);
+```
+
+- **Upload Files** (see [Assets]())
+```html
+<input type="file" id="photo" accept="image/*" />
+```
+```js
+const BuildingRecord = skygear.Record.extend('building');
+const saxonTower = new BuildingRecord({
+    photo: new skygear.Asset({
+        name: 'Saxon Tower.jpg',
+        file: document.getElementById('photo').files[0]
+    })
+});
+skygear.publicDB.save(saxonTower);
+```
+
+- **Manage Social Relations between Users** (see [Social Relations]())
+```js
+const bob = new skygear.User({ user_id: '523e7df5-655c-4250-8df0-c38271ea53ba' });
+const friend = new skygear.relation.Friend([bob]);
+skygear.relation.add(friend);
```

Comment:
need comment to explain that we have added a friend relationship between the currently logged-in user and Bob